### PR TITLE
[FLINK-34051][checkpoint] Fix equals/hashCode/toString for SavepointRestoreSettings

### PR DESCRIPTION
**Pull request by masteryhx**
## What is the purpose of the change

SavepointRestoreSettings#equals/hashCode/toString missed restoreMode property

## Brief change log

  - Add restoreMode property for SavepointRestoreSettings

## Verifying this change

This change added tests and can be verified as follows:

  - Added SavepointRestoreSettingsTest

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java [hunk 1]
```diff
@@ -105,12 +105,14 @@ public boolean equals(Object o) {
...
         return allowNonRestoredState == that.allowNonRestoredState
-                && (Objects.equals(restorePath, that.restorePath));
+                && Objects.equals(restorePath, that.restorePath)
+                && Objects.equals(restoreMode, that.restoreMode);
     }
 
     @Override
     public int hashCode() {
         int result = restorePath != null ? restorePath.hashCode() : 0;
+        result = 31 * result + (restoreMode != null ? restoreMode.hashCode() : 0);
```
Comment (ljz2051) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java:115]: The restoreMode property in SavepointRestoreSettings should be @Nonnull ?

Comment (masteryhx) [reply flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java]: Thanks for the suggestion, I think it's reasonable to mark it as '@ Nonnull'

**Comment by ljz2051** [comment]
Thanks for the update, LGTM

**Review by ljz2051 — APPROVED** [review]
Thanks for the update, LGTM

**Pull request closed without merging**